pkg/volume/awsebs/aws_ebs_block.go

```diff
@@ -68,6 +68,11 @@ func getVolumeSpecFromGlobalMapPath(volumeName string, globalMapPath string) (*v
 	if !strings.Contains(vID, "vol-") {
 		return nil, fmt.Errorf("failed to get volumeID from global path=%s, invalid volumeID format = %s", globalMapPath, vID)
 	}
+	vID, err := formatVolumeID(volumeName, vID)
```

Comment:
With `globalMapPath=/var/lib/kubelet/plugins/kubernetes.io/aws-ebs/volumeDevices/aws:/us-east-1d/vol-03c60877cf4a627e2`, 
```
vID := filepath.Base(globalMapPath)
```
is `vol-03c60877cf4a627e2`, omitting whole `aws:/us-east-1d`.

Reply:
I ended up with this:

```
       pluginDir := plugin.host.GetVolumeDevicePluginDir(awsElasticBlockStorePluginName)
       if !strings.HasPrefix(globalMapPath, pluginDir) {
               return nil, fmt.Errorf("volume symlink %s is not in global plugin directory", globalMapPath)
       }
       fullVolumeID := strings.TrimPrefix(globalMapPath, pluginDir)
       fullVolumeID = strings.TrimLeft(fullVolumeID, "/")
       vID, err := formatVolumeID(volumeName, fullVolumeID)
```
+ made getVolumeSpecFromGlobalMapPath a method of the plugin to be able to get `host`.

Reply:
sgtm, I will copy you